src/content/content_manager.cc

```diff
@@ -351,7 +351,10 @@ std::deque<std::shared_ptr<GenericTask>> ContentManager::getTasklist()
         return taskList;
 
     taskList.push_back(std::move(t));
-    std::copy_if(taskQueue1.begin(), taskQueue1.end(), std::back_inserter(taskList), [](auto&& task) { return task->isValid(); });
+
+    for (auto&& task : taskQueue1)
+        if (task->isValid())
+            taskList.push_back(std::move(task));
```

Comment:
This moves the contents of taskQueue1 to taskList, which is not used in case taskQueue2 still contains valid entries. In this case the former entries of taskQueue1 are lost.